Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/frame/DartVmServiceEvaluatorOnLibrary.java

```diff
@@ -0,0 +1,117 @@
+package com.jetbrains.lang.dart.ide.runner.server.vmService.frame;
+
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import com.intellij.xdebugger.XSourcePosition;
+import com.intellij.xdebugger.evaluation.ExpressionInfo;
+import com.intellij.xdebugger.evaluation.XDebuggerEvaluator;
+import com.jetbrains.lang.dart.ide.runner.server.vmService.DartVmServiceDebugProcess;
+import com.jetbrains.lang.dart.psi.DartFile;
+import com.jetbrains.lang.dart.psi.DartPartOfStatement;
+import gnu.trove.THashSet;
+import org.dartlang.vm.service.element.Isolate;
+import org.dartlang.vm.service.element.LibraryRef;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+
+public class DartVmServiceEvaluatorOnLibrary extends XDebuggerEvaluator {
+  @NotNull private final DartVmServiceDebugProcess myDebugProcess;
+  @NotNull private final String myIsolateId;
+  @NotNull private final VirtualFile myFile;
+  @NotNull private final Project myProject;
+
+  public DartVmServiceEvaluatorOnLibrary(@NotNull final DartVmServiceDebugProcess debugProcess,
+                                         @NotNull final String isolateId,
+                                         @NotNull final VirtualFile file,
+                                         @NotNull final Project project) {
+    myDebugProcess = debugProcess;
+    myIsolateId = isolateId;
+    myFile = file;
+    myProject = project;
+  }
+
+  @Override
+  public boolean isCodeFragmentEvaluationSupported() {
+    return true;
```

Comment:
It's about multi-line field for the evaluated expression. Do you think we need it? If so, should it be enabled in `DartVmServiceEvaluator` as well?

Reply:
I think the multi-line field is nice for entering complex expressions Dart VM handles multi line expressions fine although it could be a bit smarter. 
For example the following works:
![image](https://user-images.githubusercontent.com/1226812/40331423-2e0b7cc6-5d05-11e8-98e0-43e809be5a98.png)
But the following currently generates an error.
![image](https://user-images.githubusercontent.com/1226812/40331451-49249510-5d05-11e8-960c-527581083299.png)
I'm also fine with keeping multi-line off until the second case is also supported.